be1-go/channel/chirp/mod.go

```diff
@@ -0,0 +1,377 @@
+package chirp
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"github.com/rs/zerolog"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/sign/schnorr"
+	"golang.org/x/xerrors"
+	"popstellar/channel"
+	generalChirping "popstellar/channel/generalChirping"
+	"popstellar/channel/inbox"
+	"popstellar/crypto"
+	jsonrpc "popstellar/message"
+	"popstellar/message/answer"
+	"popstellar/message/messagedata"
+	"popstellar/message/query"
+	"popstellar/message/query/method"
+	"popstellar/message/query/method/message"
+	"popstellar/network/socket"
+	"popstellar/validation"
+	"strconv"
+)
+
+const msgID		 = "msg id"
+
+// NewChannel returns a new initialized individual chirping channel
+func NewChannel(channelPath string, ownerKey string, hub channel.HubFunctionalities,
+				generalChannel *generalChirping.Channel, log zerolog.Logger) Channel {
```

Comment:
Not sure I understood, does it mean that general would not implement Channel anymore but only implement this new GeneralChirpFunctionalities interface? So I can remove publish, subscribe, unsubscribe?

Reply:
No, general remains the same. The only change is the type received by the function. Instead of receiving `generalChannel generalChirping.Channel`, it would receive a more specific type: a `generalChannel channel.Broadcastable` interface (better than `GeneralChirpFunctionalities` in fact). When you call `NewChannel` you can pass it your general channel since it implements `Broadcast`.